ddtrace/contrib/dogpile_cache/lock.py

```diff
@@ -0,0 +1,32 @@
+import dogpile
+
+from ...pin import Pin
+
+
+def _wrap_lock_ctor(func, instance, args, kwargs):
+    """
+    This seems rather odd. But to track hits, we need to patch the wrapped function that
+    dogpile passes to the region and locks. Unfortunately it's a closure defined inside
+    the get_or_create* methods themselves, so we can't easily patch those.
+    """
+    func(*args, **kwargs)
+    ori_backend_fetcher = instance.value_and_created_fn
+
+    def wrapped_backend_fetcher():
+        pin = Pin.get_from(dogpile.cache)
+        if not pin or not pin.enabled():
+            return ori_backend_fetcher()
+
+        hit = False
+        expired = True
+        try:
+            value, createdtime = ori_backend_fetcher()
+            hit = value is not dogpile.cache.api.NO_VALUE
+            # dogpile sometimes returns None, but only checks for truthiness. Coalesce
+            # to minimize APM users' confusion.
+            expired = instance._is_expired(createdtime) or False
+            return value, createdtime
+        finally:
+            pin.tracer.current_span().set_tag('hit', hit)
+            pin.tracer.current_span().set_tag('expired', expired)
```

Comment:
`test_traces_get_or_create_multi` is failing on its check for`hit` in the partial case. From running locally with print debug statements, it seems as if the order of the cache gets for the multi case is not always in order of the arguments passed in. To deal with the multi case, we might want to instead to set the `hit` tag based on what's already been added to the current span, e.g.: 

```python
span = pin.tracer.current_span()
span.set_tag('hit', (span.get_tag('hit') is None and hit) or (asbool(span.get_tag('hit')) and hit))
```

Something similar might be necessary for `expired`.

Reply:
Thanks for digging into that for me!

Reply:
I believe I've fixed the issue. Reran the test multiple times locally and haven't seen the failure yet.